CMakeLists.txt

```diff
@@ -0,0 +1,136 @@
+#
+# Copyright (C) 2024 Renesas Electronics Corporation.
+# Copyright (C) 2024 EPAM Systems, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cmake_minimum_required(VERSION 3.19)
+
+project("aos_core_sm_cpp")
+
+# ######################################################################################################################
+# Options
+# ######################################################################################################################
+
+option(WITH_TEST "build with test" OFF)
+option(WITH_COVERAGE "build with coverage" OFF)
+option(WITH_DOC "build with documenation" OFF)
+
+message(STATUS)
+message(STATUS "${CMAKE_PROJECT_NAME} configuration:")
+message(STATUS "CMAKE_BUILD_TYPE              = ${CMAKE_BUILD_TYPE}")
+message(STATUS "CMAKE_INSTALL_PREFIX          = ${CMAKE_INSTALL_PREFIX}")
+message(STATUS)
+message(STATUS "WITH_TEST                     = ${WITH_TEST}")
+message(STATUS "WITH_COVERAGE                 = ${WITH_COVERAGE}")
+message(STATUS "WITH_DOC                      = ${WITH_DOC}")
+message(STATUS)
+
+set(WITH_SM_API ON)
```

Comment:
Not right place. Probably need to be moved before `add_subdirectory(external/aos_core_api)`. Or if keep here need to add comment that this option is related to aos_core_api.